sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -88,7 +101,8 @@ private[sql] case class JDBCRelation(
 
   override val needConversion: Boolean = false
 
-  override val schema: StructType = JDBCRDD.resolveTable(url, table, properties)
+  override val schema: StructType =
+    JDBCRDD.resolveTable(url, table, JDBCRelation.getEffectiveProperties(properties))
```

Comment:
Yes, you are right, I removed the getEffectiveProperties() and all tests passed. This may be attributed to changes in getConnectionFactory()